pool/mock_queue_test.go

```diff
@@ -43,12 +46,32 @@ func (q *QueueTester) Put(ctx context.Context, j amboy.Job) error {
 	case <-ctx.Done():
 		return ctx.Err()
 	case q.toProcess <- j:
+		q.mutex.Lock()
+		defer q.mutex.Unlock()
+
 		q.storage[j.ID()] = j
 		return nil
 	}
 }
 
+func (q *QueueTester) Save(ctx context.Context, j amboy.Job) error {
+	q.mutex.Lock()
+	defer q.mutex.Unlock()
+
+	id := j.ID()
+	if _, ok := q.storage[id]; !ok {
```

Comment:
I read this as, "If it's not already in storage, don't store it." That seems wrong.

Reply:
save shouldn't save jobs that aren't tracked (use put for that) and put shouldn't save jobs that are tracked (use put.) I think this is correct.

Reply:
the production implementations return an error here, but in practice for in memory queues you don't need to call save because the job is always the same object.